modules/core/04-channel/keeper/events.go

```diff
@@ -83,3 +83,57 @@ func EmitTimeoutPacketEvent(ctx sdk.Context, packet exported.PacketI, channel ty
 		),
 	})
 }
+
+// EmitSendPacketEvent emits an vent with Packet data along with other packet information for relayer
+// to pick up and relay to other chain
+func EmitSendPacketEvent(ctx sdk.Context, packet exported.PacketI, channel types.Channel, timeoutHeight exported.Height) {
```

Comment:
I would move this function to the top of the file, just before `EmitRecvPacketEvent`; and `EmitWriteAcknowledgementEvent` just before `EmitAcknowledgePacketEvent`. So that the order of definition of the functions follows the expected order of the events: send, receive, write acknowledgement, process acknowledgement, and last timeout.